### PR TITLE
Add tolerance to handle floating point representation errors.

### PR DESCRIPTION
**Pull request by vaishakp**
Allow for the cosine of the polarization angle to marginally deviate from +- 1.

### CODE REVIEW

File: nrcatalogtools/lvc.py [hunk 1]
```diff
@@ -969,6 +969,18 @@ def get_nr_to_lal_rotation_angles(
     # salpha = corb_phase * n_dot_theta - sorb_phase * ln_cross_n_dot_theta
     calpha = corb_phase * n_dot_psi - sorb_phase * ln_cross_n_dot_psi
 
+    if abs(calpha) > 1:
+        calpha_err = abs(calpha) - 1
+        if calpha_err < tol:
+            print(
+                f"Correcting the polarization angle for finite precision error {calpha_err}"
+            )
+            calpha = calpha / abs(calpha)
```
Comment (prayush) [inline nrcatalogtools/lvc.py:978]: You're using `tol = 1e-6` here for a cosine, and `5e-3` in line 934 above when checking for sine squared. The overflow in latter should be smaller than the former, yet the tolerances are in the opposite order. Why?
Also, where you use `5e-3` please define it as a const variable at the top.
Preferably, have all tolerances you use be user inputs with sensible defaults.

Comment (vaishakp) [reply nrcatalogtools/lvc.py:978]: I did notice this. I decided to retain the different tolerances used here as used in LAL, just to be consistent, except in the case of calpha. We could make them user inputs, which I was waiting for untill testing completes.

Comment (prayush) [reply nrcatalogtools/lvc.py:978]: Okay, being consistent with LAL is a good reason. I'd still suggest making these tolerances as inputs with defaults as chosen by LAL. It ensures that users are aware of these tolerances. Ideally we could move constants like these to a `config.py` and read from there, but that is for another day.

Comment (vaishakp) [reply nrcatalogtools/lvc.py:981]: Sure

**Review by prayush — APPROVED** [review]

**Pull request merged**